### PR TITLE
Load system font from file

### LINKED ISSUE #1084: Windows: amethyst::ui::get_default_font isn't implemented for Path cases yet
Currently when you try to use [get_default_font](https://www.amethyst.rs/doc/master/doc/amethyst_ui/fn.get_default_font.html) on Windows(presumably on Linux too?) you get [panic](https://github.com/amethyst/amethyst/blob/master/amethyst_ui/src/font/default.rs#L13-L15)

It could be nice maybe at least fallback to builtin font and print warning log?

If I understood `font-kit` sources correctly it will be always `Path` on windows https://github.com/pcwalton/font-kit/blob/master/src/sources/directwrite.rs#L79-L86
So essentially this API should not be available on windows as of now

### PR DESCRIPTION
**Pull request by DoumanAsh**
Closes  #1084

### CODE REVIEW

File: amethyst_ui/src/font/default.rs [hunk 1]
```diff
@@ -7,23 +10,45 @@ use {
...
         Ok(handle) => match handle {
-            FontKitHandle::Path { .. } => unimplemented!(
-                "Default system font was provided as a path, this is not yet supported.
-				If you see this message, open an issue so that we know we need to implement it."
-            ),
+            FontKitHandle::Path { path, .. } => if let Some(file_name) = path.file_name().and_then(|file_name| file_name.to_str()) {
+                let format = if file_name.ends_with(".ttf") || file_name.ends_with(".otf") {
+                    Some(TtfFormat)
+                } else {
+                    warn!("System font '{}' has unknown format", file_name);
```
Comment (torkleyy) [inline amethyst_ui/src/font/default.rs]: I think this deserves `error!`.

Comment (DoumanAsh) [reply amethyst_ui/src/font/default.rs]: Ok

File: amethyst_ui/src/font/default.rs [hunk 2]
```diff
@@ -14,16 +16,34 @@ pub fn get_default_font(loader: &Loader, storage: &AssetStorage<FontAsset>) -> F
 
     match system_font {
         Ok(handle) => match handle {
-            FontKitHandle::Path { .. } => unimplemented!(
-                "Default system font was provided as a path, this is not yet supported.
-				If you see this message, open an issue so that we know we need to implement it."
-            ),
+            FontKitHandle::Path { path, .. } => {
+                if let Some(file_name) = path.file_name().and_then(|file_name| file_name.to_str()) {
+                    let format = if file_name.ends_with(".ttf") || file_name.ends_with(".otf") {
```
[the remark this answers is not in the document]
Comment (DoumanAsh) [reply amethyst_ui/src/font/default.rs]: `to_str` returns `Option<&str>`, not `String`

[the remark this answers is not in the document]
Comment (DoumanAsh) [reply amethyst_ui/src/font/default.rs]: But actually it may not be necessary to convert to `&str`

**Pull request merged**